rust-symcrypt/src/hash.rs

```diff
@@ -1,59 +1,98 @@
+/* Hashing functions. For further documentation please refer to symcrypt.h */
+
 use core::ffi::c_void;
 use std::mem;
+use std::pin::Pin;
 use std::ptr;
 use symcrypt_sys;
 
 pub const SHA256_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA256_RESULT_SIZE as usize;
 pub const SHA384_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA384_RESULT_SIZE as usize;
 
-pub trait Hash {
+/// Generic trait for stateful hashing
+///
+/// append() appends to be hashed data to the state, this operation can be done multiple times.
+///
+/// result() returns the result of the hash. Once result() is called, the lifetime of the ShaXXXState is finished.
+/// To perform other stateful hash operation you must create a new hash object via ShaXXXState::new()
+///
+/// copy() creates a copy of the current ShaXXXState
```

Comment:
```suggestion
/// Generic trait for stateful hashing
///
/// `append()` appends to be hashed data to the state, this operation can be done multiple times.
///
/// `result()` returns the result of the hash. Once `result()` is called, the lifetime of the ShaXXXState is finished.
/// To perform other stateful hash operation you must create a new hash object via ShaXXXState::new()
///
/// `copy()` creates a copy of the current ShaXXXState
```

Just adding some markdown formatting for the rustdoc pages. I think you can also write them as ``[`append`]`` and then it will also link to the docs for the `append` function.

Reply:
noted, will change to ['append`]